### PR TITLE
refactor: remove decision reports from front-end

### PR DESCRIPTION
**Pull request by OmranAbazid**
related to https://github.com/camunda/camunda-optimize/issues/13439

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #

### CODE REVIEW

**Comment by OmranAbazid** [comment]
There is a task on the parent issue to remove `reportType` completely later when it is removed from the backend.

**Comment by misiekhardcore** [comment]
in `optimize/client/e2e/utils.js`, `optimize/client/e2e/sm-tests/Collection.elements.js`, `optimize/client/e2e/sm-tests/Collection.js`  there are still some decision report functions and elements

File: optimize/client/src/modules/filter/FilterList.js [hunk 1]
```diff
@@ -387,20 +379,12 @@ FilterList.defaultProps = {
   deleteFilter: () => {},
 };
 
-function isDecisionVariable(type) {
-  return ['inputVariable', 'outputVariable'].includes(type);
-}
-
 function getVariableLabel(nameOrId, type, filterType, variables) {
```
Comment (misiekhardcore) [inline optimize/client/src/modules/filter/FilterList.js:382]: I think we dont need this function, we can keep just the `getProcessVariableLabel`

File: optimize/client/src/modules/filter/FilterList.js [hunk 2]
```diff
@@ -387,20 +379,12 @@ FilterList.defaultProps = {
...
-  return isDecisionVariable(filterType)
-    ? getDecisionVariableLabel(nameOrId, filterType, variables)
-    : getProcessVariableLabel(nameOrId, type, variables);
+  return getProcessVariableLabel(nameOrId, type, variables);
 }
 
 function variableExists(nameOrId, type, filterType, variables) {
```
Comment (misiekhardcore) [inline optimize/client/src/modules/filter/FilterList.js:386]: same for this function

File: optimize/client/src/modules/components/ReportRenderer/visualizations/Number.js [hunk 3]
```diff
@@ -105,16 +105,14 @@ export function Number({report, formatter, mightFail}) {
           } else if (measure.property === 'percentage') {
             viewString = t('report.percentageOfInstances');
           } else {
-            const config = reportConfig[reportType];
+            const config = reportConfig;
```
Comment (misiekhardcore) [inline optimize/client/src/modules/components/ReportRenderer/visualizations/Number.js:108]: do we need this variable?

File: optimize/client/src/modules/components/ReportDetails/DiagramModal.tsx [hunk 4]
```diff
@@ -56,12 +51,7 @@ export function DiagramModal({definition, type, open, onClose, mightFail}: Diagr
 
 export default withErrorHandling(DiagramModal);
 
-function loadXML(reportType: string, definition: Definition) {
+function loadXML(definition: Definition) {
```
Comment (misiekhardcore) [inline optimize/client/src/modules/components/ReportDetails/DiagramModal.tsx:54]: is this function needed?

File: optimize/client/src/components/Reports/controlPanels/Configuration/Configuration.js [hunk 5]
```diff
@@ -160,7 +160,7 @@ export default class Configuration extends Component {
               <ShowInstanceCount
                 showInstanceCount={configuration.showInstanceCount}
                 onChange={this.updateConfiguration}
-                label={reportType === 'decision' ? 'evaluation' : 'instance'}
+                label="instance"
```
Comment (misiekhardcore) [inline optimize/client/src/components/Reports/controlPanels/Configuration/Configuration.js:163]: do we need this prop?

**Review by misiekhardcore — APPROVED** [review]
pre-approving. I just found few more things

File: optimize/client/src/modules/services/reportService.js [hunk 6]
```diff
@@ -94,8 +94,7 @@ export function isCategoricalBar(report) {
 export function isCategorical({groupBy, distributedBy}) {
   return (
     ['flowNodes', 'userTasks', 'assignee', 'candidateGroup'].includes(groupBy?.type) ||
-    (['variable', 'inputVariable', 'outputVariable'].includes(groupBy?.type) &&
-      ['Boolean', 'String'].includes(groupBy.value.type)) ||
+    (groupBy?.type === 'variable' && ['Boolean', 'String'].includes(groupBy.value.type)) ||
```
[the remark this answers is not in the document]
Comment (OmranAbazid) [reply optimize/client/src/modules/services/reportService.js:97]: Yes

File: optimize/client/src/modules/services/reportConfig/reportConfig.js [hunk 7]
```diff
@@ -15,16 +15,9 @@ import {getVariableLabel} from 'variables';
...
-const reportOptionsMap = {
-  process: processOptions,
-  decision: decisionOptions,
-};
-
-export function createReportUpdate(reportType, report, updateType, newValue, payloadAdjustment) {
-  const reportOptions = reportOptionsMap[reportType];
-  let newPayload = reportOptions[updateType].find(({key}) => key === newValue).payload(report);
+export function createReportUpdate(report, updateType, newValue, payloadAdjustment) {
+  let newPayload = processOptions[updateType].find(({key}) => key === newValue).payload(report);
```
[the remark this answers is not in the document]
Comment (OmranAbazid) [reply optimize/client/src/modules/services/reportConfig/reportConfig.js:20]: I will just call it ReportOptions

**Review by misiekhardcore — APPROVED** [review]

**Pull request merged**